src/server/pbsd_main.c

```diff
@@ -2143,11 +2145,37 @@ main(int argc, char **argv)
 		if (reap_child_flag)
 			reap_child();
 #endif	/* WIN32 */
-
+		scks = (struct priority_socks *)malloc(sizeof(struct priority_socks));
```

Comment:
Instead of allocating memory dynamically using two function calls one at this location and the other at line no #2161 you can actually club these two function calls into one if we allocate memory at the line no #2161 itself. Because by that time we actually know how much memory is indeed required.

Reply:
removed this malloc